added tls support

shufps opened the pull request:
adds tls stratum support

Pmaxsd committed:
added tls support

Pmaxsd committed:
fixed crash by only letting the stratumtask disconnect the connection, controled via flags

mutatrum commented:
Not sure if it's relevant for this PR, but for Bitaxe I had to add `CONFIG_MBEDTLS_SSL_PROTO_TLS1_3=y` in `sdkconfig.default` to get TLSv1.3 support.

shufps commented:
> CONFIG_MBEDTLS_SSL_PROTO_TLS1_3

Thx for pointing out :raised_hands:

Pmaxsd committed:
switch to esp-transport as cleaner solution

shufps committed:
UI swap button (#443)

* Add AtlasPool.io dashboard link for AtlasPool.io users (#430)

* NerdOctaxeGamma rev3.1 configuration and calibration (#435)

* delete

* Add TPS53667 voltage regulator support for NerdOctaxeGamma

Implement hardware detection via GPIO3 to support both TPS53647 and TPS53667
voltage regulators on NerdOctaxeGamma boards (rev 3.1+).

Changes:
- Add GPIO3-based VR detection with pull-down for backward compatibility
- Configure TPS53667 with 6-phase operation (240A, 24.9kΩ resistor)
- Configure TPS53647 with 4-phase operation (180A, 33.2kΩ resistor)
- Disable phase shedding for NerdOctaxeGamma to ensure all 6 phases active
- Enable phase shedding for other boards (NerdEko) for efficiency
- Extend frequency range to 700MHz for TPS53667 configuration

* Adjust frequency options and limits for NerdOctaxeGamma TPS53667

* Adjust frequency and voltage defaults for NerdOctaxeGamma TPS53667

* Add VR temperature offset for NerdOctaxeGamma TPS53667

* Intermediate steps for frequency expanded in UI selection (#440)

* Voltage range for ASICs in UI expanded (#439)

* During further testing, a problem with the silicon lottery arose. For this reason, the voltage range, especially from 800 to 1000 mhz, had to be expanded to enable stable operation of the NerdQX.

* added ui swap button

---------

Co-authored-by: Matt Weinberg <mjwein@yahoo.com>
Co-authored-by: BitMaker <105243563+BitMaker-hub@users.noreply.github.com>
Co-authored-by: Andreas <hello@plebbase.com>
Co-authored-by: Pmaxsd <pmaxsd80@gmail.com>

Pmaxsd committed:
fixed some potential memory leak
